app/models/deletion.rb

```diff
@@ -30,7 +30,7 @@ def record_metadata
   end
 
   def expire_api_memcached
-    ["deps/v1/#{rubygem}", "info/#{rubygem}", "versions", "names"].each do |key|
+    ["deps/v1/#{rubygem}", "versions", "names"].each do |key|
```

Comment:
I believe in the rare condition the `yanked_info_checksum` will be set from cache. I hope this what you are pointing it:
```
* gem yank is requested for `foo`         |
* `info/foo` cache gets be deleted        |
                                          |  * client requests `info/foo`
                                          |  * `info/foo` gets calculated from DB using the state
                                          |     before `gem yank` (since `destroy` is running in transaction)
                                          |     and cache is set
* `rubygem.compact_index_info` gets       |
  called, however now it will end up      |
  reading from cache                      |
```